index.js

```diff
@@ -125,7 +131,7 @@ const queryDns = (version, options) => {
 	})();
 
 	promise.cancel = () => {
-		socket.cancel();
+		socket.destroy();
```

Comment:
What's the difference between `.cancel()` and `.destroy()`?

Reply:
Well, `socket.cancel()` without argument does nothing because it needs the query id: https://github.com/mafintosh/dns-socket/blob/master/index.js#L217
In that case, doing "promise.cancel()" was basically no-op.

Now we are doing `socket.destroy()` and checking for `socket.destroyed` so we stop unnecessary queries.